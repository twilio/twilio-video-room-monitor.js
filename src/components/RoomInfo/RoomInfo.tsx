import React from 'react';
import Datum from '../typography/Datum/Datum';
import Headline from '../typography/Headline/Headline';
import useDominantSpeaker from '../../hooks/useDominantSpeaker/useDominantSpeaker';
import useIsRecording from '../../hooks/useIsRecording/useIsRecording';
import useRoom from '../../hooks/useRoom/useRoom';
import useRoomState from '../../hooks/useRoomState/useRoomState';
import useStats from '../../hooks/useStats/useStats';

export default function RoomInfo() {
  const room = useRoom();
  const dominantSpeaker = useDominantSpeaker();
  const roomState = useRoomState();
  const isRecording = useIsRecording();
  const { currentReceivedBitrate, currentSentBitrate } = useStats();

  if (!room) return null;

  return (
    <div style={{ marginBottom: '1em' }}>
      <Headline>Room information:</Headline>
      {roomState === 'disconnected' ? (
        <span>Not connected to a Twilio Video room.</span>
      ) : (
        <>
          <Datum label="Room Name" value={room.name} />
          <Datum label="SID" value={room.sid} />
          <Datum label="State" value={roomState} />
          <Datum label="Dominant Speaker" value={String(dominantSpeaker?.identity || null)} />
          <Datum label="Media Region" value={room.mediaRegion} />
          <Datum label="Is Recording" value={String(isRecording)} />
          <Datum label="Total Sent Bandwidth" value={String(currentSentBitrate?.toLocaleString()) + 'kbps'} />
          <Datum label="Total Received Bandwidth" value={String(currentReceivedBitrate?.toLocaleString()) + 'kbps'} />
        </>
      )}
    </div>
  );
}
