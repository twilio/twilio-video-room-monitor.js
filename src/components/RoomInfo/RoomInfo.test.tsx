import React from 'react';
import { render } from '@testing-library/react';
import RoomInfo from './RoomInfo';
import useRoomState from '../../hooks/useRoomState/useRoomState';

jest.mock('../../hooks/useDominantSpeaker/useDominantSpeaker', () => () => ({
  dominantSpeaker: { identity: 'testIdentity' },
}));
jest.mock('../../hooks/useIsRecording/useIsRecording', () => () => 'false');
jest.mock('../../hooks/useRoom/useRoom', () => () => ({
  name: 'test123',
  sid: 'XXXXXXXXXXXXX1234',
  mediaRegion: 'testRegion',
}));
jest.mock('../../hooks/useRoomState/useRoomState');

const mockUseRoomState = useRoomState as jest.Mock<any>;
mockUseRoomState.mockImplementation(() => 'connected');

describe('The RoomInfo component', () => {
  it('should render information about the room if connected to a Twilio Video room', () => {
    const { container } = render(<RoomInfo />);
    expect(container).toMatchSnapshot();
  });

  it('should display warning when not connected to a Twilio Video room', () => {
    mockUseRoomState.mockImplementationOnce(() => 'disconnected');
    const { getByText } = render(<RoomInfo />);
    expect(getByText('Not connected to a Twilio Video room.')).toBeTruthy();
  });
});
