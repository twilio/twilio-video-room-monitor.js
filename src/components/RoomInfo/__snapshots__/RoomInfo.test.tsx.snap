// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RoomInfo component should render information about the room if connected to a Twilio Video room 1`] = `
<div>
  <div>
    <h4
      class="sc-gsTCUz fIfjRg"
    >
      Room information:
    </h4>
    <div>
      <span
        class="sc-bdfBwQ cJyRjB"
      >
        Room Name
        : 
        test123
      </span>
    </div>
    <div>
      <span
        class="sc-bdfBwQ cJyRjB"
      >
        SID
        : 
        XXXXXXXXXXXXX1234
      </span>
    </div>
    <div>
      <span
        class="sc-bdfBwQ cJyRjB"
      >
        State
        : 
        connected
      </span>
    </div>
    <div>
      <span
        class="sc-bdfBwQ cJyRjB"
      >
        Dominant Speaker
        : 
        null
      </span>
    </div>
    <div>
      <span
        class="sc-bdfBwQ cJyRjB"
      >
        Media Region
        : 
        testRegion
      </span>
    </div>
    <div>
      <span
        class="sc-bdfBwQ cJyRjB"
      >
        Is Recording
        : 
        false
      </span>
    </div>
    <div>
      <span
        class="sc-bdfBwQ cJyRjB"
      >
        Processors
        : 
        undefined
      </span>
    </div>
  </div>
</div>
`;
